### PR TITLE
Add round-trip test to JsonSerializer with overflow that needs to be escaped.

### PR DESCRIPTION
**Pull request by ahsonkhan**
cc @steveharter, @bartonjs

### CODE REVIEW

File: src/System.Text.Json/tests/Serialization/ExtensionDataTests.cs [hunk 1]
```diff
@@ -541,6 +541,35 @@ public static void DeserializeIntoJsonElementProperty()
...
         }
 
+        [Fact]
+        public static void SerializerOutputRoundtripsWhenEscaping()
+        {
+            string jsonString = "{\"\u6C49\u5B57\":\"abc\",\"Class\":{\"\u6F22\u5B57\":\"xyz\"},\"\u62DC\u6258\":{\"\u62DC\u6258\u62DC\u6258\":1}}";
+
+            ClassWithEscapedProperty input = JsonSerializer.Deserialize<ClassWithEscapedProperty>(jsonString);
+
+            Assert.Equal("abc", input.汉字);
```
Comment (ahsonkhan) [inline src/System.Text.Json/tests/Serialization/ExtensionDataTests.cs]: What can we do about property name accesses like these? This is no longer ASCII.

Comment (bartonjs) [reply src/System.Text.Json/tests/Serialization/ExtensionDataTests.cs]: You seem to have already made the member declaration be \u6C49\u5B57, theoretically you can do the same here...

```suggestion
            Assert.Equal("abc", input.\u6C49\u5B57);
```

Comment (ahsonkhan) [reply src/System.Text.Json/tests/Serialization/ExtensionDataTests.cs]: That's hilarious. I should have tried that, but was relying on auto-complete, which obviously shows the un-escaped name.

File: src/System.Text.Json/tests/Serialization/WriteValueTests.cs [hunk 2]
```diff
@@ -25,5 +25,41 @@ public static void CanWriteValueToJsonArray()
...
+        {
+            public int pizza { get; set; }
+            public int hello\u6C49\u5B57 { get; set; }
+            public int normal { get; set; }
+        }
+
+        [Fact]
+        public static void SerializeToWriterRoundTripEscaping()
+        {
+            const string jsonIn = " { \"p\\u0069zza\": 1, \"hello\\u6C49\\u5B57\": 2, \"normal\": 3 }";
```
Comment (ahsonkhan) [inline src/System.Text.Json/tests/Serialization/WriteValueTests.cs]: Hmm... GitHub doesn't like this for some reason.
image

**Comment by ahsonkhan** [comment]
Unrelated test failure from System.Xml.Linq.SDMSample.Tests on Linux Build musl_x64_Debug CI leg (netcoreapp-Linux-Debug-x64-(Alpine.39.Amd64.Open)ubuntu.1604.amd64.).

Looks like some infra related timeout. cc @dotnet/dnceng 
https://dev.azure.com/dnceng/public/_build/results?buildId=336515&view=ms.vss-test-web.build-test-results-tab&runId=9915882&paneView=debug&resultId=102620

https://helix.dot.net/api/2019-06-17/jobs/26a9f540-65be-4667-9cd3-4a62aab0ee29/workitems/System.Xml.Linq.SDMSample.Tests/console
```text
2019-09-04 05:09:57,800: INFO: 139770397326184: run(82): main: Main thread starting 10 workers
Worker 0: starting...
Worker 1: starting...
Worker 2: starting...
Worker 3: starting...
Worker 4: starting...
Worker 5: starting...
Worker 6: starting...
Worker 7: starting...
2019-09-04 05:09:57,803: INFO: 139770397326184: run(89): main: Beginning reading of test results.
2019-09-04 05:09:57,803: INFO: 139770397326184: run(95): main: Uploading results in batches of size 1000
Worker 9: starting...
Worker 8: starting...
2019-09-04 05:09:57,806: INFO: 139770397326184: run(100): main: Main thread finished queueing batches
Searching '/root/helix/work/workitem/..' for log files
Generated log list: <ul></ul>
Searching '/root/helix/work/workitem' for test results files
Found results file /root/helix/work/workitem/testResults.xml with format xunit
Worker 0: uploaded 110 results
2019-09-04 05:09:58,410: INFO: 139770397326184: run(104): main: Main thread exiting
+ date -u '+%FT%TZ'
2019-09-04T05:09:58Z
+ python /root/helix/work/correlation/xunit-reporter.py
2019-09-04 05:09:58,671: INFO: xunit-reporter(72): main: Uploading results from /root/helix/work/workitem/testResults.xml
2019-09-04 05:09:58,672: INFO: azure_storage(198): _upload: Uploading single blob: 'testResults.xml'
2019-09-04 05:09:58,715: INFO: saferequests(87): request_with_retry: Response complete with status code '201'
2019-09-04 05:09:58,716: INFO: xunit-reporter(86): main: Sending completion event
2019-09-04 05:09:58,716: INFO: event(42): send: Sending event type XUnitTestResult
2019-09-04 05:09:59,150: INFO: saferequests(87): request_with_retry: Response complete with status code '201'
+ exit 0

...
[EXECUTION TIMED OUT]
Exit Code:-3Executor timed out after 900 seconds and was killed
```

**Comment by MattGal** [comment]
@ahsonkhan not infra-related here, in this case if you review the [outer log](https://helix.dot.net/api/2019-06-17/jobs/26a9f540-65be-4667-9cd3-4a62aab0ee29/workitems/System.Xml.Linq.SDMSample.Tests/files/e4ed60fb-4375-487c-82e2-2c1d3d7631b7.log), the command which ran in the container simply did not exit. 

We have seen issues before with VSTS test reporting not completing, but that was done something like 14 minutes before the timeout occurs.   The most common cause of this sort of issue is starting another process which doesn't exit and the bash script in question therefore not actually returning.

**Comment by MattGal** [comment]
Digging a little deeper, this is the only time in history this has happened as well, so I don't think it merits further investigation until it has a repro.

**Pull request merged**